builtins.go

```diff
@@ -251,7 +250,7 @@ func builtinTrace(i *interpreter, x value, y value) (value, error) {
 	filename := trace.loc.FileName
```

Comment:
We should probably use trace.loc.File.DiagnosticFilename. The difference is that trace.loc.Filename should only contain filenames which are valid importer paths while DiagnosticFilename may have dummy strings (like the one provided in AnonymousSnippet).